contracts/LiquidDemocracy.sol

```diff
@@ -56,6 +57,59 @@ contract LiquidDemocracy {
 
         votersData[voterAddress].weight = 1;
         votersData[voterAddress].registered = true;
+        delegations[voterAddress] = voterAddress;
+    }
+
+    // Delegations
+
+    /**
+     * Delegate the voting power of a voter to another voter
+     * Only the person calling this function has the power of delegating his/her vote
+     * @param representative voter to delegate the voting power to
+     */
+    function delegate(address representative) public {
+
+        address representee = msg.sender;
+        require(delegations[representee] == representee); // At the moment you can apply a delegation once
+        require(votersData[representee].registered);
+        require(votersData[representative].registered);
+
+        delegations[representee] = representative; // Delegate my own vote
+
+        address nextRepresentative = representative;
+        address nextRepresentee = representee;
+
+        // While there is voter that hasn't performed a delegation to another voter
+        // transitively add votes
```

Comment:
I think looping through the `mapping` is okay for this basic version, but not efficient enough in the long term. This has got me thinking about what else we could try, I suppose this is exactly the point of starting simple. Let's keep it in mind as we start researching token integration, we must learn what is already out there in terms of efficiency, curious to hear your thoughts on this.

Reply:
Yes, this will happen when starting simple. I'm not sure it can get more efficient than this though (at least with the tools we know), you need to update every voter's voting power in the delegation chain in the end, and we don't expect the delegation chains to be so long. So I would say the complexity is in the order of the number of the delegations in a chain (searching in hashmap is basically constant time, you need to do this n times, 1 per each delegation of the modified chain) The drawback from this approach (compared to a previous one that I did for Delegations.sol) is that it seems difficult to handle cycles (A -> B, B- > C. C-> A) at least from a fast look at it.